plugins/modules/aws_s3.py

```diff
@@ -302,25 +302,20 @@ class Sigv4Required(Exception):
 
 
 def key_check(module, s3, bucket, obj, version=None, validate=True):
-    exists = True
     try:
         if version:
             s3.head_object(Bucket=bucket, Key=obj, VersionId=version)
         else:
             s3.head_object(Bucket=bucket, Key=obj)
-    except botocore.exceptions.ClientError as e:
-        # if a client error is thrown, check if it's a 404 error
-        # if it's a 404 error, then the object does not exist
-        error_code = int(e.response['Error']['Code'])
-        if error_code == 404:
-            exists = False
-        elif error_code == 403 and validate is False:
-            pass
-        else:
+    except is_boto3_error_code('404'):
+        return False
+    except is_boto3_error_code('403') as e:
+        if validate is True:
             module.fail_json_aws(e, msg="Failed while looking up object (during key check) %s." % obj)
-    except botocore.exceptions.BotoCoreError as e:
+    except (botocore.exceptions.BotoCoreError, botocore.exceptions.BotoCoreError) as e:
```

Comment:
Should the second one here be ClientError?

Reply:
Good catch, yes.